### PR TITLE
Fix Redis metadata parsing and reporting

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Fixes the shared Redis metadata parsing and reporting of metadata fields.

### CODE REVIEW

File: internal/component/redis/redis.go [hunk 1]
```diff
@@ -72,12 +75,32 @@ type RedisClient interface {
...
+func ParseClientFromProperties(properties map[string]string, componentType metadata.ComponentType) (client RedisClient, settings *Settings, err error) {
+	settings = &Settings{}
+
+	// upgrade legacy metadata properties
+	if componentType == metadata.StateStoreType || componentType == metadata.LockStoreType {
+		// Apply legacy defaults
+		settings.RedisMaxRetries = 3
+		settings.RedisMinRetryInterval = Duration(2 * time.Second)
+		// Parse legacy keys
+		if _, ok := properties["redisMinRetryInterval"]; !ok {
```
Comment (ItalyPaleAle) [inline internal/component/redis/redis.go:87]:
```suggestion
		if properties["redisMinRetryInterval"] == "" {
```

File: internal/component/redis/redis.go [hunk 2]
```diff
@@ -72,12 +75,32 @@ type RedisClient interface {
...
+			if val, ok := properties["maxRetryBackoff"]; ok {
```
Comment (ItalyPaleAle) [inline internal/component/redis/redis.go:88]:
```suggestion
			if properties["maxRetryBackoff"] != "" {
```

File: internal/component/redis/redis.go [hunk 3]
```diff
@@ -72,12 +75,32 @@ type RedisClient interface {
...
+				// due to different duration formats, do not simply change the key name
+				parsedVal, parseErr := strconv.ParseInt(val, 10, 0)
+				if parseErr != nil {
+					return nil, nil, fmt.Errorf("redis store error: can't parse maxRetryBackoff field: %s", parseErr)
+				}
+				settings.RedisMinRetryInterval = Duration(time.Duration(parsedVal))
+			}
+		}
+		if _, ok := properties["redisMaxRetries"]; !ok {
```
Comment (ItalyPaleAle) [inline internal/component/redis/redis.go:97]:
```suggestion
		if properties["redisMaxRetries"] == "" {
```

File: internal/component/redis/redis.go (hunk 1 above)
Comment (ItalyPaleAle) [reply internal/component/redis/redis.go:87]: In here and below, present isn't relevant as much as it is being empty or not empty.

File: internal/component/redis/redis.go [hunk 4]
```diff
@@ -72,12 +75,32 @@ type RedisClient interface {
...
+			if _, ok := properties["maxRetries"]; ok {
```
Comment (ItalyPaleAle) [inline internal/component/redis/redis.go:98]:
```suggestion
			if properties["maxRetries"] != "" {
```

File: internal/component/redis/redis.go [hunk 5]
```diff
@@ -72,12 +75,32 @@ type RedisClient interface {
...
+	// upgrade legacy metadata properties
+	if componentType == metadata.StateStoreType || componentType == metadata.LockStoreType {
+		// Apply legacy defaults
+		settings.RedisMaxRetries = 3
+		settings.RedisMinRetryInterval = Duration(2 * time.Second)
+		// Parse legacy keys
+		if properties["redisMinRetryInterval"] == "" {
+			if properties["maxRetryBackoff"] != "" {
+				// due to different duration formats, do not simply change the key name
+				parsedVal, parseErr := strconv.ParseInt(val, 10, 0)
```
Comment (tmacam) [inline internal/component/redis/redis.go:90]:
```suggestion
				parsedVal, parseErr := strconv.ParseInt(properties["maxRetryBackoff"], 10, 0)
```

**Review by ItalyPaleAle — DISMISSED** [review]

**Comment by berndverst** [comment]
Somehow this PR makes redis pubsub cert test fail. Investigating.

**Review by DeepanshuA — APPROVED** [review]

**Pull request merged**